openshift/console
# ODC-7210: Create Serverless function using the Import from the Git with Builder Images

**Fixes**: 
https://issues.redhat.com/browse/ODC-7210

**Solution Description:**

Added **Serverless Function** Import Strategy to the Git Import Form. Users now can create Serverless Functions directly from the Git Import Form if the Git Repository contains func.yaml file.

**ACs Completed**: 

- [x] Detect func.yaml in Git repo when user enter the Git URL
- [x] Read runtime, builder, buildEnvs and run envs value from the func.yaml. func.yaml
- [x] Proceed with this flow if the builder value is s2i
- [x] Allow users to view and possibly edit the environment variables which are shown in func.yaml. Add the buildEnvs and run envs to the resource 
- [x] if func.yaml is detected change the strategy to Serverless function and do not show builder images in to and  provide a section which allows the user to select the runtime version
- [x] Do not show the Pipeline and Resources section. Default to Serverless Deployment
- [x] Add label function.knative.dev: 'true' to KSVC to visualize the Serverless function in the Topology


**Screen shots / Gifs for design review**: 


https://user-images.githubusercontent.com/47265560/209876416-422b671f-e0ad-4cd7-b35f-701af4ff83fe.mp4


**Unit test coverage report**: 

WIP

**Test setup:**

1. Install the Serverless Operator
2. Go to the Git Import Add Form.
3. Paste a Git URL having func.yaml file. (e.g. https://github.com/Lucifergene/oc-func)
4. You will see the Serverless Function Strategy get selected and the Runtime and version is visible.



**Browser conformance**:
